src/test/kotlin/org/rust/ide/inspections/RsBareTraitObjectsInspectionTest.kt

```diff
@@ -49,6 +50,15 @@ class RsImplicitTraitObjectInspectionTest : RsInspectionsTestBase(RsImplicitTrai
         trait Y where Self: Foo {}
     """)
 
+    fun `test allow bare_trait_objects`() = checkWarnings("""
```

Comment:
Let's add a test with `#[allow(rust_2018_idioms)]` to check that we correctly handle this group as well